### PR TITLE
Multihop fixes

### PR DESCRIPTION
**Pull request by psaradev**

### CODE REVIEW

**Comment by ValarDragon** [comment]
Whats the state of this PR atm?

**Comment by psaradev** [comment]
> Whats the state of this PR atm?

This PR is ready on my side. After @Thunnini  make changes for estimator functions and enable CLI test, all will work fine. (could require few modifications in params)

**Comment by ValarDragon** [comment]
I'm personally fine with disabling the CLI for multihop. The important part is it exists well so that UI's can use it.

File: x/gamm/keeper/swap.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func (k Keeper) SwapExactAmountOut(
 	}
 
 	if tokenInAmount.GT(tokenInMaxAmount) {
-		return sdk.Int{}, sdk.Dec{}, sdkerrors.Wrapf(types.ErrLimitMinAmount, "%s token is larger than max amount", outRecord.Token.Denom)
+		return sdk.Int{}, sdk.Dec{}, sdkerrors.Wrapf(types.ErrLimitMaxAmount, "%s token is larger than max amount", outRecord.Token.Denom)
```
Comment (ValarDragon) [inline x/gamm/keeper/swap.go:153]: I'm going to cherry-pick this into a separate PR, just so we can keep x/gamm up to date on main

**Pull request merged**